### ISSUE TITLE
Feature: Torsten (pharmacometrics function library)

### ISSUE THREAD

**Issue by charlesm93** [stan-dev/math#314]
#### Summary:
Torsten is a prototype library of C++ functions that facilitate applications of Stan in pharmacometrics.

#### Description:
The main reason for posting the prototype is to facilitate collaboration with the Stan developer community and pursue a conversation on the best ways to create applications of Stan in pharmacometrics. 

All the functions are stored in the torsten directory under math/stan/math/. The only already existing file in stan-math that has been modified is math/stan/math/mat.hpp. 

For exposing the functions to Stan, see issue on stan-dev/stan.  

The current prototype has the following features:
* One and Two Compartment Model functions, that compute solutions analytically
* General Compartment Model functions, that compute solutions numerically
* A flexible mechanism that handles the event schedule of clinical trials

Other features are being developed and tested. 

#### Additional Information:
I still need to work through some of the pre-requisites before creating a pull request (testing, documentation, and code quality). I have used these functions on a few examples -- that is fitting models to data with Stan -- and will post these in example-models. 

I also intend to create a wiki to pool ideas.  

#### Current Version:
v2.10.0 
(read this is the version with which Torsten is compatible and with which I have been testing it).

**Comment by syclik** [comment]
@charlesm93, is there any action you need anyone to take?

**Comment by charlesm93** [comment]
@syclik, no action required. I created this issue to be consistent with Stan practices when implementing a new feature. My understanding is that this precedes submitting a pull request. Btw, I could not create a new branch directly on stan-dev, so I first created a fork -- I don't believe this should pose any problem. 

I anticipate we will be revising the design for Torsten, and pharma applications in general, during the upcoming meeting with the Stan group and Sebastian Weber. So I'm holding off designing unit tests until that's settled.
